### ISSUE TITLE
DJs cannot Edit Shows, Playlists, Overrides

### ISSUE THREAD

**Issue by tonyzeoli** [netmix/radio-station#65]
**Describe the bug**
I got a notification from my station that DJs can no longer edit Shows, Playlists, or Overrides. They can only view Shows, Playlists, Overrides. They need to be able to Edit Show pages. I checked on the demo site and this is true.

**To Reproduce**
1. Add a DJ user role
2. Login as the DJ
3. View the Radio Station admin menu
4. See there is no ability to edit a show, playlist, or override.

**Expected behavior**
A DJ should be able to edit a show, add a playlist, or schedule an override.

**Screenshots**
<img width="1552" alt="Screen Shot 2019-10-02 at 9 41 38 AM" src="https://user-images.githubusercontent.com/4053819/66050856-f0d62100-e4fb-11e9-98fe-ffb7ccb8d1fb.png">

<img width="1552" alt="Screen Shot 2019-10-02 at 9 45 36 AM" src="https://user-images.githubusercontent.com/4053819/66050867-f5023e80-e4fb-11e9-84d0-d068b2727065.png">

**Desktop (please complete the following information):**
n/a

**Smartphone (please complete the following information):**
n/a

**Additional context**
I'm thinking this was always the intended functionality, but somehow before we updated the plugin DJs could edit this stuff. They certainly have the ability to add/edit a Post and assign that Post to the Show, but they can't edit the Show page itself.

**Comment by tonyzeoli** [comment]
@majick777 When installing the Members plugin, it displays the Show Edit allow but under Schedule Overrides and not under Shows or Playlists

**Comment by tonyzeoli** [comment]
@majick777 I just got a second support request on this issue - major big so let’s try and address this.

**Comment by tonyzeoli** [comment]
@MikeNGarrett @majick777 It seems some people are having problems with Radio Station wherease the DJ can no longer edit shows, playlists, overrides, etc... I'm not sure what happened here and I'm working on troubleshooting by reverting back to WordPress 5.2.2 to see if that's the issue whereas WordPress 5.2.3 may have impacted something, but I'm not seeing that on the demo site when I had my host revert. So, I'm not sure what's happened here with any coding updates to update to the latest WordPress coding standards, but something is wrong.

**Comment by MikeNGarrett** [comment]
It looks like setting `in_array` to strict causes these checks to fail in some places.

**Comment by tonyzeoli** [comment]
@MikeNGarrett @majick777 Okay, I think I jumped the gun here in some respect too. Nikki, the original developer, requires the user to download Members to open up edit of the Show page. I rolled back to an earlier version of Radio Station on WPVMFM.org, installed the Member plugin, and logged in as a DJ assigned to a Show and not I can see that I CAN edit the show. But if you're seeing an error, Mike, in the `in_array` and, I'll pull this down and test it all out on the demo site now.

**Comment by MikeNGarrett** [comment]
The PR is still required.

**Comment by tonyzeoli** [comment]
@MikeNGarrett Sorry, can you explain "the PR is still required." Are you saying that the Member plugin is still be required to edit the Show? That's what I'm gathering from your post, but I just need clarity. Thanks.

**Comment by MikeNGarrett** [comment]
No. What I mean is that the code changes I submitted are still applicable in order to ensure DJs can edit their own shows.

**Comment by majick777** [comment]
Sorry I missed this new issue as was busy forging ahead on other enhancements/features. 

I think the strict typecasting for `in_array` checks introduced in 2.2.6 is overkill, it has created this bug and without testing further may have introduced other ones. So I've reverted all these (by removing the 3rd argument from all `in_array` checks) for 2.2.8 release to be safe. They could be re-introduced on a case by case basis if really wanted - so long as there is testing done at the same time. But I think  this kind of typecasting is too strict and is going to cause more hassle than it's worth.

So I've pushed 2.2.8 release with minor bugfixes including this one. I was hoping to get the other enhancements in this release but it's worth pushing these fixes first as they aren't quite done yet.

**Comment by tonyzeoli** [comment]
@majick777 @MikeNGarrett Okay, I think we've resolved all of the issues based on Tony's fixed. This issue blocked the DJ from editing a show page and blocked blog posts from displaying under Blogs. I haven't yet found anything else, but will continue testing. I should have tested in greater detail by walking through all of the functions of the plugin before releasing the last version. Tony and I communicated this week and I've pushed all the changes in the 2.2.8 release.
